source/dubregistry/registry.d

```diff
@@ -275,6 +304,64 @@ class DubRegistry {
 		return ret;
 	}
 
+	PackageInfo[string] getPackageInfosRecursive(string[] packnames, PackageInfoFlags flags)
+	{
+		import dub.recipe.packagerecipe : getBasePackageName;
+
+		PackageInfo[string] infos;
+		void[0][string] visited;
+		foreach (packname; packnames)
+		{
+			logDebug("getPackageInfosRecursive: %s", packname);
+			getPackageInfosRecursive(packname, flags, infos, visited);
+		}
+		logDebug("getPackageInfosRecursive for %s returned %s", packnames, infos.byKey);
+		return infos;
+	}
+
+	private void getPackageInfosRecursive(string packname, PackageInfoFlags flags, ref PackageInfo[string] infos, ref void[0][string] visited)
+	{
+		import dub.recipe.packagerecipe : getBasePackageName, getSubPackageName;
+		import std.range : dropOne;
+		import std.algorithm.searching : find;
+
+		if (packname in visited)
+			return;
+		visited[packname] = typeof(visited[packname]).init;
+
+		auto basepkg = () @trusted {
+			return getBasePackageName(packname);
+		}();
```

Comment:
Apparently this regressed from @safe to @system from dub 1.3 to dub 1.7
Fix: https://github.com/dlang/dub/pull/1432